cloudferry_devlab/cloudferry_devlab/generate_load.py

```diff
@@ -265,15 +266,21 @@ def update_filtering_file(self):
             vm_id_list = [x.__dict__['id'] for x in
                           self.novaclient.servers.list()]
 
-            img_id_list = [x.__dict__['id'] for x in
-                           self.glanceclient.images.list(search_opts={
+            img_name_list = [x.__dict__['name'] for x in
+                             self.glanceclient.images.list(search_opts={
                                'is_public': False})]
 
             for key in filter_dict.keys():
                 if key == 'images':
-                    for img_id in img_id_list:
-                        if img_id not in not_incl_img:
-                            filter_dict[key]['images_list'].append(str(img_id))
+                    for img_name in img_name_list:
+                        if (img_list == 'exclude_images_list' and img_name
```

Comment:
It's too hard to understand this logic, please re-write and make it more simple to read. Also optimize to not use the duplicated validations like "img_list == 'image_list'"